Week-1/session_1/web_intro.md

```diff
@@ -0,0 +1,62 @@
+Different websites can be stored on the same or different web servers but that doesn’t affect the actual website that you are seeing in your computer. The web server can be any software or hardware but is usually a software running on a computer. One web server can handle multiple users at any given time which is a necessity otherwise there had to be a web server for each user and considering the current world population, is nearly close to impossible. A web server is never disconnected from the internet because if it was, then it won’t be able to receive any requests, and therefore cannot process them.
+
+- Apache
+- Nginx
+- Microsoft IIS
+- Lighttpd
+- Jigsaw
+#
+#
+# 3.What is web hosting, and what are the different types of hosting services available for websites?
+
+##### Web hosting is a service that allows you to publish your website on the internet. It involves storing all your website files on a server and providing a way for people to access those files through a web browser. Think of it like renting a space on the internet.
+- Free Hosting
+- Virtual/Shared Hosting
+- Dedicated Hosting
+- Co-location Hosting
+
+
+# 4.What is scaling, and why is it important for web applications? How does scaling differ for vertical and horizontal scaling?
+
+##### Scaling alters the size of a system. In the scaling process, we either compress or expand the system to meet the expected needs. The scaling operation can be achieved by adding resources to meet the smaller expectation in the current system, by adding a new system to the existing one, or both. 
+#
+#
+# 1. Vertical Scaling: 
+##### When new resources are added to the existing system to meet the expectation, it is known as vertical scaling. 
+##### Consider a rack of servers and resources that comprises the existing system. (as shown in the figure). Now when the existing system fails to meet the expected needs, and the expected needs can be met by just adding resources, this is considered vertical scaling. Vertical scaling is based on the idea of adding more power(CPU, RAM) to existing systems, basically adding more resources.
+##### Vertical scaling is not only easy but also cheaper than Horizontal Scaling. It also requires less time to be fixed. 
+
+# 2.Horizontal Scaling:
+##### When new server racks are added to the existing system to meet the higher expectation, it is known as horizontal scaling. 
+##### Consider a rack of servers and resources that comprises the existing system. (as shown in the figure). Now when the existing system fails to meet the expected needs, and the expected needs cannot be met by just adding resources, we need to add completely new servers. This is considered horizontal scaling. Horizontal scaling is based on the idea of adding more machines to our pool of resources. Horizontal scaling is difficult and also costlier than Vertical Scaling. It also requires more time to be fixed. 
+#
```

Comment:
Refer https://www.cloudzero.com/blog/horizontal-vs-vertical-scaling for more details and better understanding.